rcp/org.testeditor.rcp4.views.tests/src/org/testeditor/rcp4/views/tests/tree/AmlModelTreeAdapterTest.xtend

```diff
@@ -0,0 +1,117 @@
+package org.testeditor.rcp4.views.tests.tree
+
+import javax.inject.Inject
+import org.junit.Before
+import org.junit.Test
+import org.mockito.MockitoAnnotations
+import org.testeditor.rcp4.views.AmlModelTreeAdapter
+
+class AmlModelTreeAdapterTest extends AbstractParserTest {
+
+	@Inject
+	AmlModelTreeAdapter amlModelTreeAdapter
+
+	@Before
+	override void setUp() {
+		super.setUp
+		MockitoAnnotations.initMocks(this);
+	}
+
+	@Test
+	def void amlModelHasComponentsInTree() {
+		// given
+		val input = '''
+			package pa
+					
+			component MyApp { }
+			component MyApp2 { }
+		'''
+		val model = parser.parse(input)
+
+		// when
+		val children = amlModelTreeAdapter.children(model).toSet
+
+		// then
+		assertEquals(children, model.components.toSet)
+		assertSize(children, 2)
+	}
+
+	@Test
+	def void componentElementHasInteractionsInTree() {
+		// given
+		val input = '''
+			package pa
+					
+			component MyApp {
+			 	element MyButton is Button { }
+			}
+			
+			element type Button {
+				interactions = push, release
+			}
+			
+			interaction type push { }
+			
+			interaction type release { }
+		'''
+		val model = parser.parse(input)
+
+		// when
+		val app = model.components.findFirst[name == "MyApp"]
+		val children = amlModelTreeAdapter.children(app.elements.findFirst[name == "MyButton"]).toSet
+
+		// then
+		assertEquals(children, model.interactionTypes.toSet)
+		assertSize(children, 2)
+	}
+
+	@Test
+	def void componentHasOwnAndInheritedInheritedInteractionsInTree() {
+		// given
+		val input = '''
+			package pa
+			
+			component other is KillApplication { 
+				element MyInheritedButton is Button { }
+			}
+			
+			component MyApp is Application includes other { 
+				element MyButton is Button { }
+			}
+			
+			component type Application {
+				interactions = open, close
+			}
+			
+			component type KillApplication {
+				interactions = kill
+			}
+			
+			element type Button {
+				interactions = push, release
+			}
+			
+			interaction type open { }
+			
+			interaction type close { }
+			
+			interaction type kill {	}
+			
+			interaction type push { }
+			
+			interaction type release { }
+		'''
+		val model = parser.parse(input)
+
+		// when
+		val children = amlModelTreeAdapter.children(model.components.findFirst[it.name == "MyApp"]).toSet
```

Comment:
`model.components.findFirst[it.name == "MyApp"]` should be assigned to a variable in the `// given` part and checked to be not-null.